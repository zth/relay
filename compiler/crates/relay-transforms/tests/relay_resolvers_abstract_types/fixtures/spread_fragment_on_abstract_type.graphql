# relay-resolver-enable-interface-output-type

query spreadFragmentOnAbstractTypeQuery {
    cat {
        ...spreadFragmentOnAbstractTypeFragment
    }
}

fragment spreadFragmentOnAbstractTypeFragment on Cat {
    description
}

# %extensions%

interface Cat {
    description: String
}

type Tabby implements Cat {
    description: String
}

type Persian implements Cat {
    description: String
}

extend type Query {
    cat: Cat
}
