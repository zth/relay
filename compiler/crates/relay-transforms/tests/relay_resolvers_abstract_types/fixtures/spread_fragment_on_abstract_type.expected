==================================== INPUT ====================================
# relay-resolver-enable-interface-output-type

query spreadFragmentOnAbstractTypeQuery {
    cat {
        ...spreadFragmentOnAbstractTypeFragment
    }
}

fragment spreadFragmentOnAbstractTypeFragment on Cat {
    description
}

# %extensions%

interface Cat {
    description: String
}

type Tabby implements Cat {
    description: String
}

type Persian implements Cat {
    description: String
}

extend type Query {
    cat: Cat
}
==================================== OUTPUT ===================================
fragment spreadFragmentOnAbstractTypeFragment on Cat {
  ... on Tabby {
    description
  }
  ... on Persian {
    description
  }
}

query spreadFragmentOnAbstractTypeQuery {
  cat {
    ...spreadFragmentOnAbstractTypeFragment
  }
}
